webui/react/src/components/UPlot/UPlotChart.tsx

```diff
@@ -149,135 +97,247 @@ const UPlotChart: React.FC<Props> = ({
 }: Props) => {
   const chartRef = useRef<uPlot>();
   const chartDivRef = useRef<HTMLDivElement>(null);
-  const boundsRef = useRef<Record<string, ChartBounds>>({});
+  const zoomScalesRef = useRef<ZoomScales>({});
+  const dataRef = useRef<uPlot.AlignedData>(EMPTY_DATA);
+  const [ isEmpty, setIsEmpty ] = useState(true);
   const [ isReady, setIsReady ] = useState(false);
+  const resize = useResize(chartDivRef);
 
-  const hasData = data && data.length > 1 && (options?.mode === 2 || data?.[0]?.length);
-
-  const extendedOptions = useMemo(
-    () =>
-      getExtendedOptions(
-        options,
-        boundsRef,
-        chartDivRef.current?.offsetWidth,
-        () => setIsReady(true),
-        () => setIsReady(false),
-      ),
-    [ options ],
-  );
-  const previousExtendedOptions = usePrevious(extendedOptions, undefined);
+  const fullOptions = useMemo(() => {
+    return uPlot.assign({
+      hooks: {
+        destroy: [ () => setIsReady(false) ],
+        init: [ (u: uPlot) => {
+          u.over.ondblclick = () => {
```

Comment:
I wasn't sure about that either, but ultimately went with this just because that's what the [uPlot demos (by the author)](https://github.com/leeoniya/uPlot/blob/master/demos/zoom-fetch.html#L37) implemented... Probably makes more sense to just use `cursor.bind`.

Reply:
initially i thought it might have the undesired effect of triggering if someone double clicks on the margins or legends or axes of the charts, but it seems to work as intended.